modules/dispatch/src/lib.rs

```diff
@@ -102,6 +102,13 @@ pub struct MessagePayload<SourceChainAccountId, TargetChainAccountPublic, Target
 	pub weight: Weight,
 	/// Call origin to be used during dispatch.
 	pub origin: CallOrigin<SourceChainAccountId, TargetChainAccountPublic, TargetChainSignature>,
+	/// If true, then the sender has decided to pay dispatch fee at the target chain.
+	///
+	/// The fee is paid right before the message is dispatched. So in case of any other
+	/// issues (like invalid call encoding, invalid signature, ...) we won't do any direct
+	/// transfers. Instead, we're returning fee related to this message dispatch to the
+	/// relayer.
+	pub pay_dispatch_fee_at_target_chain: bool,
```

Comment:
To make it more future proof I'd opt for an enum:

```suggestion
	pub dispatch_fee:  DispatchFee::{SourceChain, TargetChain},
```

this will allow us to add other options in the future (like paying with non-primary token) without breaking backward compatibility of payload format.